.jenkinsci/artifacts.groovy

```diff
@@ -11,17 +11,26 @@ def uploadArtifacts(filePaths, uploadPath, artifactServers=['artifact.soramitsu.
   }
   def shaSumBinary = 'sha256sum'
   def md5SumBinary = 'md5sum'
+  def gpgKeyBinary = 'gpg --armor --detach-sign --batch --yes --passphrase-fd 0'
   if (agentType == 'Darwin') {
     shaSumBinary = 'shasum -a 256'
     md5SumBinary = 'md5 -r'
   }
   sh "> \$(pwd)/batch.txt"
-  filePathsConverted.each {
-    sh "echo put ${it} $uploadPath >> \$(pwd)/batch.txt;"
-    sh "$shaSumBinary ${it} | cut -d' ' -f1 > \$(pwd)/\$(basename ${it}).sha256"
-    sh "$md5SumBinary ${it} | cut -d' ' -f1 > \$(pwd)/\$(basename ${it}).md5"
-    sh "echo put \$(pwd)/\$(basename ${it}).sha256 $uploadPath >> \$(pwd)/batch.txt;"
-    sh "echo put \$(pwd)/\$(basename ${it}).md5 $uploadPath >> \$(pwd)/batch.txt;"
+
+  withCredentials([file(credentialsId: 'ci_gpg_privkey', variable: 'CI_GPG_PRIVKEY'), string(credentialsId: 'ci_gpg_masterkey', variable: 'CI_GPG_MASTERKEY')]) {
+
+    sh "gpg --import ${CI_GPG_PRIVKEY}"
```

Comment:
This one fails on MacOS with the following error: `error sending to agent: Inappropriate ioctl for device`. [There is a proposed solution](https://github.com/keybase/keybase-issues/issues/2798) for this issue, but it didn't help. It then fails with an error: `error sending to agent: No such file or directory`. 
I didn't check the import on Linux though. Make sure it works on Linux too.

Reply:
I found solution on this page: https://myshittycode.com/2017/08/07/maven-gpg-plugin-prevent-signing-prompt-or-gpg-signing-failed-no-such-file-or-directory-error/

Summary: added "--pinentry-mode loopback" arguments on signing